src/catchup/CatchupRange.cpp

```diff
@@ -37,4 +39,45 @@ operator!=(CatchupRange const& x, CatchupRange const& y)
 {
     return !(x == y);
 }
+
+CatchupRange
+makeCatchupRange(ResolvedCatchupConfiguration const& configuration,
+                 HistoryManager const& historyManager)
+{
+    auto toCheckpoint =
+        historyManager.nextCheckpointLedger(configuration.toLedger() + 1) - 1;
+
+    auto firstNotKnownCheckpoint = [&]() {
+        auto lcl = configuration.lastClosedLedger();
+        auto nextCheckpoint = historyManager.nextCheckpointLedger(lcl + 1) - 1;
+        if (lcl == nextCheckpoint)
+        {
+            return nextCheckpoint + historyManager.getCheckpointFrequency();
+        }
+        return nextCheckpoint;
+    }();
+
+    auto catchupFromLedger =
+        configuration.toLedger() > configuration.count()
+            ? configuration.toLedger() - configuration.count()
+            : 0;
+    auto catchupFromCheckpoint =
+        historyManager.nextCheckpointLedger(catchupFromLedger) - 1;
+
+    // ensure that we have at least configuration.count() ledgers if
+    // possible
+    if (catchupFromLedger < catchupFromCheckpoint &&
+        catchupFromCheckpoint > historyManager.getCheckpointFrequency())
+    {
+        catchupFromCheckpoint -= historyManager.getCheckpointFrequency();
+    }
+
+    auto fromCheckpoint =
```

Comment:
is this correct (computation of fromCheckpoint)?
It seems that it should be catchupFromCheckpoint if we have computed it successfully otherwise default to firstNotKnownCheckpoint : I think there is an edge case when firstNotKnownCheckpoint  is in the future (which will cause the local node to wait for the next checkpoint instead of downloading data from the previous checkpoint).